### PR TITLE
feat(openbmc): Create a new OpenBMC brand and utilize a common brand

### PR DESCRIPTION
**Pull request by jskazinski**
The main purpose of this new feature is to pave the way for OpenBMC command support. This is only a start at supporting OpenBMC commands and a common brand was created where commands are the same for all Redfish Services.

Release Notes:

- Created new 'common' command folder. Brand is searched first, then common.
- Added new 'show brands' command which displays all brands and a command count.
- Moved many commands from 'systems' and 'example' to 'common'
- Minor bug correction in 'create session'
- Created a new Open BMC brand (obmc) as a starting point

Signed-off-by: Joe Skazinski <joseph.skazinski@seagate.com>

-----
[View rendered CHANGELOG.md](https://github.com/Seagate/SystemsRedfishPy/blob/feat/openbmc/CHANGELOG.md)
[View rendered DESIGN.md](https://github.com/Seagate/SystemsRedfishPy/blob/feat/openbmc/DESIGN.md)
[View rendered README.md](https://github.com/Seagate/SystemsRedfishPy/blob/feat/openbmc/README.md)
[View rendered UNITTEST.md](https://github.com/Seagate/SystemsRedfishPy/blob/feat/openbmc/UNITTEST.md)

### CODE REVIEW

**Review by mjenglish — APPROVED** [review]
- show_brands.py
- create_account.py
- get_accounts.py
- redfishCommand.py

Overall, this looks good and I approve. There are some smaller comments that I'll make during our meeting today.

**Review by scott-howe-1 — APPROVED** [review]

**Review by mjenglish — CHANGES_REQUESTED** [review]

File: commands/obmc/create_account.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    command = ''
+
+    @classmethod
+    def prepare_url(self, redfishConfig, command):
+        self.command = command
+        return (RedfishSystem.get_uri(redfishConfig, 'Accounts'))
+
+    @classmethod
+    def process_json(self, redfishConfig, url):
+
```
[the remark this answers is not in the document]
Comment (scott-howe-1) [reply commands/obmc/create_account.py:63]: @mjenglish, At this point, these sort of whitespace corrections are best left to a separate, general discussion about coding standards for this project.  What we enforce and what we don't.   I don't want to unnecessarily add to the size of these PR's with back-and-forth opinions about how whitespace should be handled.  We should have that discussion once, then just enforce it via the PR's.
However, I do not see a `coding_standard.md` type file for this project yet, so we may need to add that to the todo list.

[the remark this answers is not in the document]
Comment (mjenglish) [reply commands/obmc/create_account.py:63]: These are requested changes, not required changes.

[the remark this answers is not in the document]
Comment (scott-howe-1) [reply commands/obmc/create_account.py:63]: That statement implies that I'm not allowed to have an opinion on your suggestion (which I know is **_not_** the message you intended to send).

If I think the space is fine, am I now going to have to make a comment to every one of these types of formatting comments to keep it? 
The presence of these types of formatting comments in the PR's force team members to spent time on them.  One person creating them.  Others interpreting them.  It's a waste of time and something a coding standard solves. 

One thing to keep in mind with these PR's is the difference between a `functional change` and `stylistic change`.  Your `for-loop` suggestion further down is an excellent example of a useful `functional change`.  These whitespace formatting discussions are subjective `stylistic changes` that can create unnecessary back-and-forth.

File: commands/obmc/get_accounts.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+            if ('Name' in link.jsonData):
+                self.Name = link.jsonData['Name']
+
+            # RoleId
+            if ('RoleId' in link.jsonData):
+                self.RoleId = link.jsonData['RoleId']
+
+            # UserName
+            if ('UserName' in link.jsonData):
+                self.UserName = link.jsonData['UserName']
```
[the remark this answers is not in the document]
Comment (jskazinski) [reply commands/obmc/get_accounts.py:93]: Excellent simplification of code, thank you.

File: core/redfishCommand.py [hunk 3]
```diff
@@ -63,14 +64,43 @@ def execute(cls, redfishConfig, command, echo = False):
...
             else:
+                handlerFile = 'commands/' + brand + '/' + words[0] + '_' + words[1] + '.py'
                 handlerName = 'commands.' + brand + '.' + words[0] + '_' + words[1]
 
+        # Check brand version of command
+        if (path.exists(handlerFile) == False):
+            Trace.log(TraceLevel.TRACE, '++ file ({}) does not exist, check common folder'.format(handlerFile))
+            brand = 'common'
+            if (len(words) == 1):
+                handlerFile = 'commands/' + brand + '/' + words[0] + '.py'
```
[the remark this answers is not in the document]
Comment (jskazinski) [reply core/redfishCommand.py:88]: Agreed. 'brand' should be validated when that configuration setting is updated and the command line values like word[0] should be validated here. A separate issue will be created.

File: commands/obmc/get_accounts.py [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+    def process_json(self, redfishConfig, url):
+
+        if (not url):
+            return
+
+        # GET Account Service information
+        self.accounts = []
+        self.link = UrlAccess.process_request(redfishConfig, UrlStatus(url))
+
+        # Retrieve a link to Accounts
```
[the remark this answers is not in the document]
Comment (jskazinski) [reply commands/obmc/get_accounts.py:95]: I would say that the purpose of this method is to store account data for display, which when using a Redfish API takes four steps. I would agree that improvements can be made to this function, and to all functions that do very similar code, but I request that this be done in a separate ticket. I did address the width being set to a magic number, and that function can be reused in other display functions.

**Pull request merged**